prettyPrinters/providers.py

```diff
@@ -145,7 +149,9 @@ def StdStrSummaryProvider(valobj, dict):
     process = data_ptr.GetProcess()
     data = process.ReadMemory(start, length, error)
     if error.Success():
-        return '"%s"' % data
+        if system() == "Darwin":
+            return '"%s"' % data
+        return '"%s"' % data.decode(encoding='UTF-8')
```

Comment:
I've found the cause of different behavior on osx and linux - different version of python.
It's 2 on osx and 3 on linux. And since python 2 `str` is just sequence of bytes we don't need to decode it while python 3 `str` is unicode string and we have to decode it into bytes.
So let's check python version instead of system name or find a function from stdlib that does it